mismi-ec2/src/Mismi/EC2/Data.hs

```diff
@@ -65,3 +69,23 @@ newtype SecurityGroupId =
   SecurityGroupId {
       securityGroupId :: Text
     } deriving (Eq, Show)
+
+
+data InstanceProfile =
+  InstanceProfile {
+    instanceProfileARN :: InstanceProfileARN
+  } deriving (Eq, Show)
+
+newtype InstanceProfileARN =
+  InstanceProfileARN {
+    unInstanceProfileARN :: Text
+  } deriving (Eq, Show)
+
+newtype AccountNumber = AccountNumber {
+    unAccountNumber :: Text
+  } deriving (Eq, Show)
+
+extractAccountNumber :: InstanceProfileARN -> Maybe AccountNumber
+extractAccountNumber =
+  -- arn:partition:service:region:account-id:resourcetype/resource
+  fmap AccountNumber . listToMaybe . drop 4 . splitOn ":" . unInstanceProfileARN
```

Comment:
Yep, it's a number. Looks like it's even [always twelve digits](https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html).